modules/virt-building-vm-containerdisk-image.adoc

```diff
@@ -0,0 +1,160 @@
++
+[source,terminal]
+----
+# dnf install libguestfs-tools
+----
+* You have installed the Podman CLI tool (`podman`).
+
+.Procedure
+
+. Verify that you can build a {op-system-base} 8.7 image:
++
+[source,terminal]
+----
+# composer-cli distros list
+----
++
+[NOTE]
+====
+To run the `composer-cli` commands as non-root, add your user to the `weldr` or `root` groups:
+
+[source,terminal]
+----
+# usermod -a -G weldr user
+----
+[source,terminal]
+----
+$ newgrp weldr
+----
+====
+
+. Enter the following command to create an image blueprint file in TOML format that contains the packages to be installed, kernel customizations, and the services to be disabled during boot time:
++
+[source,terminal]
+----
+$ cat << EOF > dpdk-vm.toml
+name = "dpdk_image"
+description = "Image to use with the DPDK checkup"
+version = "0.0.1"
+distro = "rhel-87"
+
+[[packages]]
+name = "dpdk"
+
+[[packages]]
+name = "dpdk-tools"
+
+[[packages]]
+name = "driverctl"
+
+[[packages]]
+name = "tuned-profiles-cpu-partitioning"
+
+[customizations.kernel]
+append = "default_hugepagesz=1GB hugepagesz=1G hugepages=8 isolcpus=2-7"
+
+[customizations.services]
+disabled = ["NetworkManager-wait-online", "sshd"]
+EOF
+----
+
+. Push the blueprint file to the image builder tool by running the following command:
++
+[source,terminal]
+----
+# composer-cli blueprints push dpdk-vm.toml
+----
+
+. Generate the system image by specifying the blueprint name and output file format. The Universally Unique Identifier (UUID) of the image is displayed when you start the compose process.
++
+[source,terminal]
+----
+# composer-cli compose start dpdk_image qcow2
+----
+
+. Wait for the compose process to complete. The compose status must show `FINISHED` before you can continue to the next step.
++
+[source,terminal]
+----
+# composer-cli compose status
+----
+
+. Enter the following command to download the `qcow2` image file by specifying its UUID:
++
+[source,terminal]
+----
+# composer-cli compose image <UUID>
+----
+
+. Create the customization scripts by running the following commands:
++
+[source,terminal]
+----
+$ cat <<EOF >customize-vm
+echo  isolated_cores=2-7 > /etc/tuned/cpu-partitioning-variables.conf
+tuned-adm profile cpu-partitioning
+echo "options vfio enable_unsafe_noiommu_mode=1" > /etc/modprobe.d/vfio-noiommu.conf
+EOF
+----
++
+[source,terminal]
+----
+$ cat <<EOF >first-boot
+driverctl set-override 0000:06:00.0 vfio-pci
+driverctl set-override 0000:07:00.0 vfio-pci
+
+mkdir /mnt/huge
+mount /mnt/huge --source nodev -t hugetlbfs -o pagesize=1GB
+EOF
+----
+
+. Customize the system image by using the `virt-customize` tool:
++
+[source,terminal]
+----
+$ virt-customize -a <UUID>.qcow2 --run=customize-vm --firstboot=first-boot --selinux-relabel
+----
+
+. To create a Dockerfile that contains all the commands to build the container disk image, enter the following command:
++
+[source,terminal]
+----
+$ cat << EOF > Dockerfile
+FROM scratch
+COPY <uuid>-disk.qcow2 /disk/
```

Comment:
Please consider adding a note about editing this line to the proper `qcow2` image name.